pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SegmentCommitterFactory.java

```diff
@@ -54,7 +57,13 @@ public SegmentCommitterFactory(Logger segmentLogger, ServerSegmentCompletionProt
   public SegmentCommitter createSegmentCommitter(SegmentCompletionProtocol.Request.Params params,
       String controllerVipUrl)
       throws URISyntaxException {
-    boolean uploadToFs = _streamConfig.isServerUploadToDeepStore();
+    InstanceDataManagerConfig instanceDataManagerConfig = _indexLoadingConfig.getInstanceDataManagerConfig();
+    PinotConfiguration config = instanceDataManagerConfig.getConfig();
```

Comment:
Add a method in `InstanceDataManagerConfig` to return whether to upload segment to deep store. You should be able to see other examples in the class

Reply:
Fixed.